### PR TITLE
Bug fixes for results

### PR DESCRIPTION
**Pull request by arpita-20**
Bug fixes for results page-

1-While deleting test on test page ,that particular test is not being
 deleted on results page 
 solution-added status column to keep track of the test page 
 2-For one Email id ,only one link should be generated
solution-fixed this bug and also added toast for this error

### CODE REVIEW

File: app/components/results/GroupByTestItems.tsx [hunk 1]
```diff
@@ -25,7 +19,7 @@ const GroupByTestItems = ({
         <div className=" col-span-1 truncate">
```
Comment (anuragP04) [inline app/components/results/GroupByTestItems.tsx:19]: remove extra space.

File: app/components/results/GroupByTestItems.tsx [hunk 2]
```diff
@@ -25,7 +19,7 @@ const GroupByTestItems = ({
...
           <span className=" text-base leading-6 text-gray-700">{index}</span>
```
Comment (anuragP04) [inline app/components/results/GroupByTestItems.tsx:20]: remove extra space.

File: app/components/results/GroupByTestItems.tsx [hunk 3]
```diff
@@ -38,9 +32,18 @@ const GroupByTestItems = ({
         <div className="col-span-2 truncate text-base leading-6 text-gray-700">
           {candidateTests.candidateTest?.length}
```
Comment (anuragP04) [inline app/components/results/GroupByTestItems.tsx:33]: candidateTests.candidateTest ??

create object such that these structure can be avoided or if it contains more information then update prop name

File: app/components/results/GroupByTests.tsx [hunk 4]
```diff
@@ -52,15 +51,18 @@ const GroupByTests = () => {
               <span className="col-span-1 text-sm font-semibold  text-gray-500">
                 {resultConstants.srno}
```
Comment (anuragP04) [inline app/components/results/GroupByTests.tsx:52]: use camelCase for variable names

File: app/components/results/GroupByTests.tsx [hunk 5]
```diff
@@ -52,15 +51,18 @@ const GroupByTests = () => {
...
               </span>
-              <span className="col-span-3 text-sm font-semibold  text-gray-500">
+              <span className="col-span-2 text-sm font-semibold text-gray-500">
                 {commonConstants.total} {resultConstants.totalAttended}
               </span>
+              <span className="col-span-1 text-sm font-semibold text-gray-500">
+                {resultConstants.status}
+              </span>
             </div>
             <div id="GroupByTestItems" className="col-span-10 grid">
```
Comment (anuragP04) [inline app/components/results/GroupByTests.tsx:67]: use camelCase or kebab-case for ID, don't use PascalCase for ID naming, preferred: kebab-case

File: app/components/tests/InviteCandidatePopup.tsx [hunk 6]
```diff
@@ -3,6 +3,7 @@ import { Fragment, useEffect, useState } from 'react'
 import { Dialog, Transition } from '@headlessui/react'
 import { Icon } from '@iconify/react'
 import { toast } from 'react-toastify'
+import { candidateExam, testsConstants } from '~/constants/common.constants'
```
Comment (anuragP04) [inline app/components/tests/InviteCandidatePopup.tsx:6]: we can make constants group/objects name more consistant, in one we are using Constant and in other not.

File: app/constants/common.constants.ts [hunk 7]
```diff
@@ -95,9 +97,12 @@ export const resultConstants = {
   totalAttended: 'Attended',
   noTestAlert: 'No test found. Add your test first',
   srno: 'Sr.No',
```
Comment (anuragP04) [inline app/constants/common.constants.ts:99]: use camelCase

File: app/constants/common.constants.ts [hunk 8]
```diff
@@ -129,7 +134,11 @@ export const candidateExam = {
...
   examTitle: 'Pre-Interview Assessment',
   info: 'Info',
-  noOfQuestions: "No. of Questions"
+  noOfQuestions: 'No. of Questions',
+  errorOfcreateCandidateTest: 'errorOfcreateCandidateTest',
+  errorofcreateCandidateData: 'errorofcreateCandidateData',
+  candidateTestCreated: 'created',
+  error: 'error',
 }
 export const cypress = {
```
Comment (anuragP04) [inline app/constants/common.constants.ts:143]: try to reduce the constants inside cypress, we don't need this one

File: app/components/tests/InviteCandidatePopup.tsx [hunk 9]
```diff
@@ -23,12 +24,18 @@ const InviteCandidatePopup = ({
     if (actionData?.status == 401) {
       toast.warn(actionData.message)
     }
-    if (actionData?.candidateInviteStatus == 'created') {
-      if (actionData?.testId === testId) toast.success('Candidates Invited')
+    if (
+      actionData?.candidateInviteStatus == candidateExam.candidateTestCreated
```
Comment (richa-copods) [inline app/components/tests/InviteCandidatePopup.tsx:28]: Please use strict check `===`

File: app/constants/common.constants.ts [hunk 10]
```diff
@@ -66,7 +66,9 @@ export const testsConstants = {
   created: 'Created',
   actionsText: 'Actions',
   noTestFound: 'No Test Found',
-  noCandidateForTest: "No Candidate for this Test"
+  noCandidateForTest: 'No Candidate for this Test',
```
Comment (richa-copods) [inline app/constants/common.constants.ts:69]: Please change the text to- `No candidate has attempted this test`

File: app/constants/common.constants.ts [hunk 11]
```diff
@@ -129,7 +134,11 @@ export const candidateExam = {
   startSection: 'Start Now',
   examTitle: 'Pre-Interview Assessment',
   info: 'Info',
-  noOfQuestions: "No. of Questions"
+  noOfQuestions: 'No. of Questions',
+  errorOfcreateCandidateTest: 'errorOfcreateCandidateTest',
```
Comment (richa-copods) [inline app/constants/common.constants.ts:138]: Please remove if not being used

File: app/components/results/GroupByTests.tsx [hunk 12]
```diff
@@ -52,15 +51,18 @@ const GroupByTests = () => {
               <span className="col-span-1 text-sm font-semibold  text-gray-500">
```
Comment (richa-copods) [inline app/components/results/GroupByTests.tsx:51]: Please remove extra space

**Review by richa-copods — CHANGES_REQUESTED** [review]

File: app/models/candidate.server.ts [hunk 13]
```diff
@@ -16,40 +22,51 @@ export async function createIndividualCandidate({ email, createdById }: { email:
...
           email,
-          createdById
+          createdById,
         },
       })
     }
     return user
   } catch (error) {
-    console.log(error)
     throw new Error('Candidate Create Error..!')
```
Comment (anuragP04) [inline app/models/candidate.server.ts:31]: use constants

File: app/models/candidate.server.ts [hunk 14]
```diff
@@ -16,40 +22,51 @@ export async function createIndividualCandidate({ email, createdById }: { email:
...
       data: {
         testId,
-        candidateId
-      }
+        candidateId,
+      },
     })
     return candidateTest
   } catch (error) {
     throw new Error('Candidate Test Create Error..!')
```
Comment (anuragP04) [inline app/models/candidate.server.ts:52]: use constant here

File: app/components/results/GroupByTestItems.tsx [hunk 15]
```diff
@@ -38,9 +32,18 @@ const GroupByTestItems = ({
...
         </div>
-        <div className="col-span-3 truncate text-base leading-6 text-gray-700">
+        <div className="col-span-2 text-base leading-6 text-gray-700">
```
Comment (yash6102000) [inline app/components/results/GroupByTestItems.tsx:35]: no need to add leading class if we r using font size classes provided by tailwind

File: app/components/results/GroupByTestItems.tsx (hunk 2 above)
Comment (yash6102000) [inline app/components/results/GroupByTestItems.tsx:20]: no need to add leading class if we r using font size classes provided by tailwind

File: app/components/results/GroupByTestItems.tsx [hunk 16]
```diff
@@ -38,9 +32,18 @@ const GroupByTestItems = ({
         <div className="col-span-2 truncate text-base leading-6 text-gray-700">
```
Comment (yash6102000) [inline app/components/results/GroupByTestItems.tsx:32]: no need to add leading class if we r using font size classes provided by tailwind

File: app/components/results/GroupByTestItems.tsx [hunk 17]
```diff
@@ -38,9 +32,18 @@ const GroupByTestItems = ({
...
           {candidateTests.candidateTest?.length}
         </div>
-        <div className="col-span-3 truncate text-base leading-6 text-gray-700">
+        <div className="col-span-2 text-base leading-6 text-gray-700">
           {candidateTests?.count}
         </div>
+        <div
+          className={`${
+            candidateTests?.deleted ? 'text-yellow-500' : 'text-green-500'
+          } col-span-1 text-base leading-6`}
```
Comment (yash6102000) [inline app/components/results/GroupByTestItems.tsx:41]: no need to add leading class if we r using font size classes provided by tailwind

**Review by anuragP04 — APPROVED** [review]

**Pull request merged**